maintenance_timesheet/models/hr_timesheet.py

```diff
@@ -16,31 +16,38 @@ def onchange_maintenance_request_id(self):
             self.project_id = self.maintenance_request_id.project_id
             self.task_id = self.maintenance_request_id.task_id
 
-    @api.model
-    def create(self, values):
-        if values.get("maintenance_request_id"):
-            self._check_request_done(values.get("maintenance_request_id"))
-        return super().create(values)
+    @api.model_create_multi
+    def create(self, vals_list):
+        maintenance_request_ids = [
+            vals.get("maintenance_request_id")
+            for vals in vals_list
+            if vals.get("maintenance_request_id")
+        ]
+        self._check_request_done(maintenance_request_ids)
+        return super().create(vals_list)
 
     def write(self, values):
         current_request = self.maintenance_request_id
         new_request_id = values.get("maintenance_request_id", False)
         if current_request:
-            self._check_request_done(current_request.id)
+            self._check_request_done([current_request.id])
         if new_request_id:
-            self._check_request_done(new_request_id)
+            self._check_request_done([new_request_id])
         return super().write(values)
 
     def unlink(self):
-        for timesheet in self.filtered(lambda x: x.maintenance_request_id):
-            self._check_request_done(timesheet.maintenance_request_id.id)
+        self._check_request_done(
+            self.filtered(lambda x: x.maintenance_request_id).maintenance_request_id.ids
+        )
         return super().unlink()
 
-    def _check_request_done(self, request_id):
+    def _check_request_done(self, request_ids):
```

Comment:
You shouldn't change the API of this method, as there can be other modules overriding it. Instead, just call N times the method.

Reply:
You are right. On the other hand this will trigger a query for each record. We should change this in the future.

Reply:
I'll propose an alternative that satisfies both worlds... I'll let me know what you think